src/main/lens-proxy/handle-request-for.injectable.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+import { getInjectable } from "@ogre-tools/injectable";
+import type { ServerResponse, IncomingMessage } from "http";
+import { apiKubePrefix } from "../../common/vars";
+import contentSecurityPolicyInjectable from "../../common/vars/content-security-policy.injectable";
+import type { ClusterContextHandler } from "../context-handler/context-handler";
+import routeRequestInjectable from "../router/route-request.injectable";
+import { getBoolean } from "../utils/parse-query";
+import getClusterForRequestInjectable from "./get-cluster-for-request.injectable";
+import type HttpProxyServer from "http-proxy";
+
+const watchParam = "watch";
+const followParam = "follow";
+
+const isLongRunningRequest = (reqUrl: string) => {
+  const url = new URL(reqUrl, "http://localhost");
+
+  return getBoolean(url.searchParams, watchParam) || getBoolean(url.searchParams, followParam);
+};
+
+export type HandleRequest = (req: IncomingMessage, res: ServerResponse) => Promise<void>;
+export type HandleRequestFor = (proxy: HttpProxyServer) => HandleRequest;
+
+const handleRequestForInjectable = getInjectable({
+  id: "handle-request-for",
+  instantiate: (di): HandleRequestFor => {
+    const getClusterForRequest = di.inject(getClusterForRequestInjectable);
+    const contentSecurityPolicy = di.inject(contentSecurityPolicyInjectable);
+    const routeRequest = di.inject(routeRequestInjectable);
+
+    const getProxyTarget = async (req: IncomingMessage, contextHandler: ClusterContextHandler) => {
+      if (req.url?.startsWith(apiKubePrefix)) {
+        delete req.headers.authorization;
+        req.url = req.url.replace(apiKubePrefix, "");
+
+        return contextHandler.getApiTarget(isLongRunningRequest(req.url));
+      }
+
+      return undefined;
+    };
+
+    return (proxy) => async (req, res) => {
```

Comment:
Would it be possible for the proxy to be a normal injected dependency instead of one injected using a configuration parameter of a "for" function (eg. `handle-request-for`).

Reply:
Will attempt to do so, but I think I did this to prevent a circular dependency.

Reply:
Unfortunately it would cause a circular dep instead I will just consolidate it.

Reply:
The circular dep does indicate a probable design error, and this kludges around it. I would advice caution :/

Reply:
Yes so I remove this injectable since it was only used in one place